src/foram3d/simulation.ts

```diff
@@ -293,6 +310,44 @@ module Foram3D {
       );
     }
 
+    private onMouseClick(event) {
+      event.preventDefault();
+
+      var chamber = this.getPointedChamber(event);
+
+      if (chamber && this._onChamberClick) {
```

Comment:
Sprawdzenie, czy `this._onChamberClick` istnieje musi być wykonane wcześniej. `getPointedChamber` jest relatywnie ciężkie, a nie ma sensu go robić, jeśli i tak nic się nie wykona. Analogicznie przy `onMouseMove`.